tasks/pkg-debian.yml

```diff
@@ -23,6 +23,11 @@
     state: present
   when: datadog_apt_key_url_new is defined
 
+- name: Remove previous datadog apt list file
+  file:
+    path: /etc/apt/sources.list.d/ansible_datadog_agent.list
+    state: absent
```

Comment:
not sure I understand what this task is for: if I understand correctly the tasks below will update the `/etc/apt/sources.list.d/ansible_datadog_agent.list` anyway, so why do we need to delete it here?

Reply:
The custom repository won't get deleted if this is not present. Moreover, if the internal variables (`datadog_agentX_yyyyy_repo`) are modified, the previous values won't get removed from the source files, which may cause issues. IMO it's safer to explicitly remove the list file.

It does have the downside of making the role not idempotent though, so this is up for discussion.

Reply:
Ok, thanks for the explanation, makes sense. Indeed, the idempotency would be nice to have, but not blocking for this PR.

In a future version, we may want to revisit using the `apt_repository` task. For our use case, I think all of this logic would be easier to implement with one single `file` task that would trigger `apt-get update` on the datadog repo whenever needed. Not necessary for now though.